server/database/db_build.sql

```diff
@@ -1,26 +1,26 @@
 BEGIN;
 
 CREATE TABLE users (
-  user_id TEXT PRIMARY KEY NOT NULL,
+  user_id TEXT NOT NULL PRIMARY KEY,
   date_created TIMESTAMP NOT NULL DEFAULT now()
 );
 
 CREATE TABLE goals (
-  goal_id TEXT PRIMARY KEY NOT NULL,
-  user_id TEXT NOT NULL,
+  goal_id TEXT NOT NULL PRIMARY KEY,
+  user_id TEXT REFERENCES users(user_id),
   title TEXT NOT NULL,
   icon TEXT,
   deleted BOOLEAN DEFAULT FALSE,
-  date_created TIMESTAMP NOT NULL DEFAULT now()
+  date_created TEXT
 );
 
 CREATE TABLE ratings (
-  rating_id SERIAL PRIMARY KEY NOT NULL,
-  user_id TEXT,
-  goal_id TEXT,
+  rating_id TEXT NOT NULL PRIMARY KEY,
+  user_id TEXT REFERENCES users(user_id),
+  goal_id TEXT REFERENCES goals(goal_id),
   rating INTEGER,
   comment TEXT,
-  date_created TIMESTAMP NOT NULL DEFAULT now()
+  date_created TEXT
```

Comment:
at least i think so... it all seems to be fine at the moment. Whatever I said before might have been wrong...